### PR TITLE
Enhance parity for SSM logic and integration tests

### PR DESCRIPTION
**Pull request by whummer**
This PR marks the tests in `test_ssm.py` as `aws_validated`, and adds minor refactorings to achieve parity with the logic in real AWS. Addresses https://github.com/localstack/localstack/issues/6551

### CODE REVIEW

File: localstack/services/ssm/provider.py [hunk 1]
```diff
@@ -21,11 +22,102 @@
...
 
 
+class ValidationException(CommonServiceException):
+    def __init__(self, message=None):
+        super().__init__("ValidationException", message=message, sender_fault=True)
+
+
+# TODO: check if _normalize_name(..) calls are still required here
 class SsmProvider(SsmApi, ABC):
+    def get_parameters(
```
Comment (whummer) [inline localstack/services/ssm/provider.py:38]: Only reordering of methods here (moving util methods to the bottom), no functional changes.

File: tests/integration/test_ssm.py [hunk 2]
```diff
@@ -33,22 +37,25 @@ def test_put_parameters(self, ssm_client, create_parameter):
...
             Name=f"/{param_a}/b/c",
             Value="123",
             Type="String",
         )
 
         _assert(f"/{param_a}/b/c", f"/{param_a}/b/c", ssm_client)
-        _assert(f"/{param_a}//b//c", f"/{param_a}/b/c", ssm_client)
-        _assert(f"{param_a}/b//c", f"/{param_a}/b/c", ssm_client)
+        for pname in [f"/{param_a}//b//c", f"{param_a}/b/c"]:
+            with pytest.raises(Exception) as exc:
```
Comment (thrau) [inline tests/integration/test_ssm.py:53]: nit: an alternative could be to parametrize the test, which makes debugging the test much easier if it fails

File: localstack/services/ssm/provider.py [hunk 3]
```diff
@@ -34,7 +126,15 @@ def _has_secrets(names: ParameterNameList) -> Boolean:
         return maybe_secret is not None
 
     @staticmethod
-    def _normalize_name(param_name: ParameterName) -> ParameterName:
+    def _normalize_name(param_name: ParameterName, validate=False) -> ParameterName:
+        if validate:
+            msg = (
```
Comment (thrau) [inline localstack/services/ssm/provider.py:131]: nit: could move this in the if block or directly in the constructor

**Review by thrau — APPROVED** [review]
LGTM! ++ For AWS validated tests 💯

File: localstack/services/ssm/provider.py (hunk 3 above)
Comment (whummer) [reply localstack/services/ssm/provider.py]: Good catch, updated 👍

**Pull request merged**